src/gui/spaces/space.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) by Fabian Müller <fmueller@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "space.h"
+#include "gui/guiutility.h"
+
+#include <QCoreApplication>
+#include <QIcon>
+
+namespace {
+static const char contextC[] = "space";
+const auto personalC = QLatin1String("personal");
+const auto shareC = QLatin1String("virtual");
+}
+
+namespace OCC::Spaces {
+
+Space::Space(const QString &name, const QString &subtitle, const QUrl &webUrl, const QUrl &webDavUrl, const QUrl &imageUrl)
```

Comment:
Any reason we need that constructor, can't we just own the Drive and operate on it?

Reply:
It gives us some flexibility with regards to (future) testing.